### ISSUE TITLE
Scipy 1.3.0 build broken in AIX

### ISSUE THREAD

**Issue by ayappanec** [scipy/scipy#10398]
Scipy 1.3.0 build is broken in AIX. It is used to work till 1.2.1
Below is the error.

  /opt/freeware/lib/python3.7/config/ld_so_aix /opt/freeware/bin/gfortran -Wall -g -bI:/opt/freeware/lib/python3.7/config/python.exp -Wall -g -shared build/temp.aix-6.1-3.7/build/src.aix-6.1-3.7/scipy/fftpack/_fftpackmodule.o build/temp.aix-6.1-3.7/scipy/fftpack/src/zfft.o build/temp.aix-6.1-3.7/scipy/fftpack/src/drfft.o build/temp.aix-6.1-3.7/scipy/fftpack/src/zrfft.o build/temp.aix-6.1-3.7/scipy/fftpack/src/zfftnd.o build/temp.aix-6.1-3.7/build/src.aix-6.1-3.7/scipy/fftpack/src/dct.o build/temp.aix-6.1-3.7/build/src.aix-6.1-3.7/scipy/fftpack/src/dst.o build/temp.aix-6.1-3.7/build/src.aix-6.1-3.7/build/src.aix-6.1-3.7/scipy/fftpack/fortranobject.o -L/opt/freeware/lib/gcc/powerpc-ibm-aix6.1.0.0/6.3.0 -L/opt/freeware/lib64 -Lbuild/temp.aix-6.1-3.7 -ldfftpack -lfftpack -lgfortran -o build/lib.aix-6.1-3.7/scipy/fftpack/_fftpack.so
  Running from scipy source directory.
  /tmp/pip-build-env-fqyrp11r/overlay/lib64/python3.7/site-packages/numpy/distutils/system_info.py:716: UserWarning: Specified path /tmp/pip-build-env-fqyrp11r/overlay/lib64/python3.7/site-packages/numpy/__init__.py/include/python3.7m is invalid.
    return self.get_paths(self.section, key)
  ld: 0711-736 ERROR: Input file build/temp.aix-6.1-3.7/build/src.aix-6.1-3.7/scipy/fftpack/_fftpackmodule.o:
        XCOFF64 object files are not allowed in 32-bit mode.
  collect2: error: ld returned 8 exit status
  ld: 0711-736 ERROR: Input file build/temp.aix-6.1-3.7/build/src.aix-6.1-3.7/scipy/fftpack/_fftpackmodule.o:
        XCOFF64 object files are not allowed in 32-bit mode.
  collect2: error: ld returned 8 exit status

The root cause is "-maix64" option (needed for 64bit compilation) is not getting passed to gfortran. This issue was there sometime back but i have a committed a patch to numpy to fix this --> https://github.com/numpy/numpy/commit/68833d7f43c631082d842f2e3999d57242a21732
And this used to work till Scipy 1.2.1 . Looks like something got changed in Scipy 1.3.0 and it no longer honors the build options from numpy ?

**Comment by tylerjereddy** [comment]
Some of us (myself included) do have access to AIX machines on the GCC compile farm for debugging, but it may take a while for us to get to this. Unfortunately, as you likely know, adding AIX / big endian nodes to i.e., Travis CI hasn't been a priority for IBM / the ecosystem thus far. Would be useful to prevent this kind of thing though.

**Comment by rgommers** [comment]
There's nothing in `fftpack/setup.py` that can explain this. I'm thinking one of two things:
- is the build invoked the same way for 1.2.1 and 1.3.0, and is the numpy version identical?
- we added `pyproject.toml` in 1.3.0, so if you're building with `pip install .` or similar, that gets invoked and may fall back to an older numpy version for the build that does not have the required distutils patch.

A full build log would be useful here.

**Comment by ayappanec** [comment]
I tested this in a clean AIX machine. Installed python3 & python3-devel that internally has setuptools 40.8.0 and pip 19.0.3. Did "python3 -m pip install numpy" which installed 1.16.4 version. Then tried to install scipy through same pip command ( it took 1.3.0 version ) and it failed because of the above mentioned issue.  Attaching the log here.
[scipy_build_log.txt](https://github.com/scipy/scipy/files/3372771/scipy_build_log.txt)

**Comment by pv** [comment]
Probably some mismatch between the gfortran and C compilers, or their compiler flags used;
likely nothing to do with fftpack specifically as it just happens to be first module to be linked.

Note that if you want the Numpy version that you manually installed to be used for building, you need to use `pip install --no-build-isolation`. The problem here may be that Numpy 1.13.3 or 1.14.5 contains too old numpy.distutils to work on aix.

**Comment by ayappanec** [comment]
I have only one numpy installed in the machine through pip which is numpy-1.16.4 version . And i could see "numpy/distutils/fcompiler/gnu.py" has the patch which i mentioned in the first thread here. 
pip install scipy==1.2.1 is successful. 
It's the latest version 1.3.0 which is failing. So there has to be something with the newer scipy which creates this issue.

**Comment by pv** [comment]
@ayappanec: please see https://pip.pypa.io/en/stable/reference/pip/#pep-517-and-518-support
What you install manually does not determine what's available at build time.

If that's the problem, could you figure out what's the oldest Numpy version that works on aix.
Then, update scipy's pyproject.toml along the lines of
```
[build-system]
requires = [
    "wheel",
    "setuptools",
    "Cython>=0.29.2",
    "numpy==1.13.3; python_version=='3.5' and platform_machine!='AIX'",
    "numpy==1.13.3; python_version=='3.6' and platform_machine!='AIX'",
    "numpy==1.14.5; python_version>='3.7' and platform_machine!='AIX'",
    "numpy==1.16.4; python_version=='3.5' and platform_machine=='AIX'",
    "numpy==1.16.4; python_version=='3.6' and platform_machine=='AIX'",
    "numpy==1.16.4; python_version>='3.7' and platform_machine=='AIX'",
    "pybind11>=2.2.4",
]
```
and send a PR.

**Comment by ayappanec** [comment]
Thanks @pv @rgommers for the information. 
The numpy patch for aix is available from version 1.16.0 onwards only. I tested the changes provided by @pv here and it is working. Will send a PR.

**Comment by ayappanec** [comment]
> 
> 
> Some of us (myself included) do have access to AIX machines on the GCC compile farm for debugging, but it may take a while for us to get to this. Unfortunately, as you likely know, adding AIX / big endian nodes to i.e., Travis CI hasn't been a priority for IBM / the ecosystem thus far. Would be useful to prevent this kind of thing though.

Yes. Having Travis CI for AIX build will be great. Let me know what things are needed for it.
